packages/accordions/src/elements/accordion/Accordion.tsx

```diff
@@ -0,0 +1,118 @@
+import { StyledAccordion } from '../../styled';
+import { AccordionContext } from '../../utils';
+import { Section } from '../accordion/components/Section';
+import { Header } from '../accordion/components/Header';
+import { Label } from '../accordion/components/Label';
+import { Panel } from '../accordion/components/Panel';
+
+interface IStaticAccordionExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Section: typeof Section;
+  Header: typeof Header;
+  Label: typeof Label;
+  Panel: typeof Panel;
+}
+
+interface IAccordionProps extends Omit<HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  /** The `aria-level` used to apply heading rank on accordion headers */
+  level: number;
+  /** A controlled prop that determines which sections are expanded */
+  expandedSections?: number[];
+  /** Hides the bottom border under each accordion panel */
+  isBare?: boolean;
+  /** Determines if panels can be collapsed in an uncontrolled accordion */
+  isCollapsible?: boolean;
+  /** Reduces padding on accordion headers and panels */
+  isCompact?: boolean;
+  /** Determines if multiple panels can be expanded at the same time in an uncontrolled accordion */
+  isExpandable?: boolean;
+  /** A callback that is invoked with an accordion section’s index when the expanded state changes */
+  onChange?: (index: number) => void;
+}
+
+/**
+ * Accepts all `<div>` attributes and events. Also accepts sub-components:
+
+ *  - `Accordion.Section`
+ *  - `Accordion.Header`
+ *  - `Accordion.Panel`
+ * 
+ * Note: The `Accordion.Label` is a sub-component of `Accordion.Header`.
+ */
+export const Accordion = forwardRef<HTMLDivElement, IAccordionProps>(
+  (
+    {
+      level,
+      isBare,
+      onChange,
+      isCompact,
+      isExpandable,
+      isCollapsible,
+      expandedSections: controlledExpandedSections,
+      ...props
+    },
+    ref
+  ) => {
+    const { expandedSections, getHeaderProps, getTriggerProps, getPanelProps } = useAccordion({
+      collapsible: isCollapsible,
+      expandable: isExpandable,
+      onChange,
+      expandedSections: controlledExpandedSections
+    });
+    const currentIndexRef = useRef(0);
+
+    useEffect(() => {
+      currentIndexRef.current = 0;
+    });
+
+    const value = {
+      level,
+      isBare,
+      isCompact,
+      getPanelProps,
+      getHeaderProps,
+      getTriggerProps,
+      currentIndexRef,
+      expandedSections
+    };
+
+    return (
+      <AccordionContext.Provider value={value}>
+        <StyledAccordion ref={ref} {...props} />
+      </AccordionContext.Provider>
+    );
+  }
+) as IStaticAccordionExport<HTMLDivElement, IAccordionProps>;
+
+Accordion.Section = Section;
+Accordion.Header = Header;
+Accordion.Label = Label;
+Accordion.Panel = Panel;
+
+Accordion.displayName = 'Accordion';
+
+Accordion.defaultProps = {
+  isBare: false,
+  isCompact: false,
+  isCollapsible: false,
```

Comment:
@steue I wonder, should we consider the `true` by default? I'm tinkering with the knobs in the examples and I'm finding my inability to collapse the accordions kind of frustrating. It seems more likely that we would want to allow the user to open and close accordions by default and then mandate them open in special cases. The functionality seems the same by default, it just give the user the ability to close the accordion to save space.

Reply:
If this is the decision, we should set the `isCollapsible` prop to `true` by default along with the demo toggle. Totally fine and makes sense to me.

Reply:
> it just gives the user the ability to close the accordion to save space.

An accordion that has one expanded section still saves space. According to WAI-ARIA, accordions are _"commonly used to reduce the need to scroll when presenting multiple sections of content on a single page"_. Here's their demo:

https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html

I'm unsure that "saving" space is a strong enough justification for having this prop be defaulted to `true`. Saving space is context-specific (how much space? what kind of space?). The nature of the accordion already saves space (for scrolling on a page). I think it is nice to have the default for `isCollapsible" as "off" to match the WAI-ARIA example and other accordion examples I've seen. However, this isn't mandatory, and Garden can have an opinionated view here.

Reply:
Furthermore, the inability to collapse on the "Advanced" controlled example doesn't map with the description provided on the `isCollapsible` component: "Determines if panels can be collapsed in an uncontrolled accordion". Setting a reasonable default might help with this.

Reply:
My thinking from a fundamental point of view is an accordion component opens and closes, so to have it unable to close by default feels odd to me considering the intended functionality. I'd defer on this particular point but giving the user the ability to close seems like a benefit with zero drawback.

Reply:
Another point to make here is that we would never show disabled in a component’s default state. Having these props for asserting control is great. But I think we almost have to set `isCollapsed = true`

Reply:
@rossmoody 

> to have it unable to close by default feels odd to me 

It's not clear to me if there are fundamental reasons for defaulting `isCollapsible` to `true` besides "it feels odd". Does the WAI-ARIA [example](https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html) feel odd too?

> but giving the user the ability to close seems like a benefit without any drawbacks

There are various implementations of Accordion behaviors — each suitable to a different UX context. Each of these implementations can be achieved with various combinations of the `isCollapsible` and `isExpandable` props. Maybe the reason for defaulting `isCollapsible` to `true` is that research has shown that our users consistently use that implementation most?

Reply:
@rossmoody 

> to have it unable to close by default feels odd to me 

It's not clear to me if there are fundamental reasons for defaulting `isCollapsible` to `true` besides "it feels odd". Does the WAI-ARIA [example](https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html) feel odd too?

> but giving the user the ability to close seems like a benefit without any drawbacks

There are various implementations of Accordion behaviors — each suitable to a different UX context. Each of these implementations can be achieved with various combinations of the `isCollapsible` and `isExpandable` props. Maybe the reason for defaulting `isCollapsible` to `true` is that research has shown that our users consistently use that implementation most? 

@jzempel 

> we would never show disabled in a component’s default state.

Could you clarify what this means? Is it the case that the basic example would never be used since it always has one panel expanded, and one header button with `aria-disabled="true"`?

Reply:
> Could you clarify what this means?

@steue is going to provide disabled styling for `isCollapsible={false}` sections. That means the most basic form of an `<Accordion>` (with no props added) would be disabled. We can't have that. It makes sense to go with Ross' design instincts on this and set the default for `isCollapsible` to `true`.

Reply:
> Does the WAI-ARIA example feel odd too?

Yes, I have always thought it did.

Reply:
> Furthermore, the inability to collapse on the "Advanced" controlled example doesn't map with the description provided on the isCollapsible component: "Determines if panels can be collapsed in an uncontrolled accordion". Setting a reasonable default might help with this.

I'll go ahead and add the ability to collapse on the "Advanced" controlled example. The `isCollapsible` prop and description doesn't apply to a controlled accordion. A controlled has custom behavior.